### PR TITLE
Created Response Schema and added Status Code constants

### PR DESCRIPTION
**Pull request by scoobhidu**
[feat] added statusCodes.go in constants and functions to create ErrorSchema

### CODE REVIEW

**Comment by scoobhidu** [comment]
@abhishekraj272 @uzaxirr created a fresh PR that one got way messy

File: constants/status_codes.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	200: "OK",
+	201: "Created",
+	202: "Accepted",
+	203: "Non-Authoritative Information",
+	204: "No Content",
+	205: "Reset Content",
+	206: "Partial Content",
+	207: "Multi-Status",
+	208: "Already Reported",
+	226: "IMUsed",
```
Comment (uzaxirr) [inline constants/status_codes.go:18]: This map should be added in `constants/constants.go` i don't think there's need to create a new file.
and i doubt if we gonna use soo many values so just limit it to the below mentioned one
```go
const HTTPStatusCodes := map[int]string{
	http.StatusOK:                   http.StatusText(http.StatusOK),
	http.StatusCreated:              http.StatusText(http.StatusCreated),
	http.StatusAccepted:             http.StatusText(http.StatusAccepted),
	http.StatusNonAuthoritativeInfo: http.StatusText(http.StatusNonAuthoritativeInfo),
	http.StatusNoContent:            http.StatusText(http.StatusNoContent),
	http.StatusBadRequest:           http.StatusText(http.StatusBadRequest),
	http.StatusUnauthorized:         http.StatusText(http.StatusUnauthorized),
	http.StatusForbidden:            http.StatusText(http.StatusForbidden),
	http.StatusNotFound:             http.StatusText(http.StatusNotFound),
	http.StatusMethodNotAllowed:     http.StatusText(http.StatusMethodNotAllowed),
	http.StatusConflict:             http.StatusText(http.StatusConflict),
	http.StatusInternalServerError:  http.StatusText(http.StatusInternalServerError),
}
```

File: helpers/response_schema.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package helpers
```
Comment (uzaxirr) [inline helpers/response_schema.go:1]: this can be added in the `utils` directory itself

**Review by uzaxirr — CHANGES_REQUESTED** [review]

**Review by uzaxirr — CHANGES_REQUESTED** [review]
Create a new directory and file `types/types.go` this file will contain the struct which can be used for both Error and Sucess Response

`types/types.go`

```go
type Response struct {
	Status  int         `json:"status"`
	Data    interface{} `json:"data"`
	Error   interface{} `json:"error"`
}

```

and then you can modify the controller, an example of what the Ping controller may look like is

```go
func Ping(c *gin.Context) {
	response := Response{
		Status: HTTPStatusCodes[http.StatusOK],
		Data:   "pong",
		Error:  struct{}{},
	}
	c.JSON(http.StatusOK, response)
}
```

It may give the below output
```json
{
    "status": 200,
    "data": "pong",
    "error": {}
}
```

And for the `/generate-presigned-url`
```go
response := Response{
	Status: HTTPStatusCodes[http.StatusOK],
	Data: map[string]interface{}{
		"preSignedURL": "https://somebucket-bucket.s3.amazonaws.com/abcdefg",
		"videoID":      "109934677.mp4",
	},
	Error: struct{}{},
}
```

and in case of error the `Data` feild will be empty while the `Error` field will have the appropriate info

**Comment by uzaxirr** [comment]
Hey @loyaltytrooper 
Working on this?

**Comment by uzaxirr** [comment]
Closing this due to inactivity

**Pull request closed without merging**